airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/integration/test_report_based_streams.py

```diff
@@ -0,0 +1,266 @@
+                assert len(output.records) == 0
+
+    @HttpMocker()
+    def test_given_report_status_cancelled_when_read_then_no_records(self, http_mocker: HttpMocker) -> None:
+        for stream_name in STREAMS.keys():
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+                http_mocker.get(
+                    _check_report_status_request(_REPORT_ID).build(),
+                    _check_report_status_response(stream_name, processing_status=ReportProcessingStatus.cancelled)
+                )
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 0
+
+    @HttpMocker()
+    def test_given_report_status_fatal_when_read_then_exception_raised(self, http_mocker: HttpMocker) -> None:
+        for stream_name in STREAMS.keys():
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+                http_mocker.get(
+                    _check_report_status_request(_REPORT_ID).build(),
+                    _check_report_status_response(stream_name, processing_status=ReportProcessingStatus.fatal)
+                )
+
+                output = self._read(stream_name, config(), expecting_exception=True)
+                assert output.errors[-1].trace.error.failure_type == FailureType.system_error
+                assert output.errors[-1].trace.error.message == f"The report for stream '{stream_name}' was not created - skip reading"
+
+    @HttpMocker()
+    def test_given_report_with_incorrect_date_format_when_read_then_formatted(self, http_mocker: HttpMocker) -> None:
+        for stream_name, params in STREAMS.items():
+            if "date_field" not in params or "expected_date_value" not in params:
+                continue
+            with self.subTest(msg=stream_name, **params):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+                http_mocker.get(_check_report_status_request(_REPORT_ID).build(), _check_report_status_response(stream_name))
+                http_mocker.get(
+                    _get_document_download_url_request(_REPORT_DOCUMENT_ID).build(), _get_document_download_url_response()
+                )
+                http_mocker.get(
+                    _download_document_request(_DOCUMENT_DOWNLOAD_URL).build(),
+                    _download_document_response(stream_name, data_format=params.get("data_format", "csv")),
+                )
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 2
+                assert output.records[0].record.data.get(params["date_field"]) == params["expected_date_value"]
```

Comment:
It is not clear to me where `expected_date_value` and what it represents from the test setup

Reply:
It is now a test's parameter